File: examples/java/core/child-contexts/risky-operation.java
public class RiskyOperationHandler extends DurableHandler<Map<String, Object>, Map> {

    @Override
    public Map handleRequest(Map<String, Object> event, DurableContext ctx) {
        Map data = (Map) event.get("data");

        var result = ctx.runInChildContext("risky_operation", Map.class, childCtx -> {
            try {
                var stepResult = childCtx.step("risky_step", Map.class,
                    stepCtx -> potentiallyFailingOperation(data));
                return Map.of("status", "success", "result", stepResult);
            } catch (Exception e) {
                // Handle error within child context
                var fallback = childCtx.step("fallback", Map.class,
                    stepCtx -> fallbackOperation(data));
                return Map.of("status", "fallback", "result", fallback, "error", e.getMessage());
            }
        });

        if ("fallback".equals(result.get("status"))) {
            return Map.of("warning", "Used fallback", "result", result.get("result"));
        }

        return result;
    }
}

File: examples/python/operations/child-contexts/risky-operation.py
@durable_with_child_context
def risky_operation(ctx: DurableContext, data: dict) -> dict:
    """Operation that might fail."""
    try:
        result = ctx.step(
            lambda _: potentially_failing_operation(data),
            name="risky_step"
        )
        return {"status": "success", "result": result}
    except Exception as e:
        # Handle error within child context
        fallback = ctx.step(
            lambda _: fallback_operation(data),
            name="fallback"
        )
        return {"status": "fallback", "result": fallback, "error": str(e)}

@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    """Handle errors in child context."""
    result = context.run_in_child_context(
        risky_operation(event["data"]),
        name="risky_operation"
    )
    
    if result["status"] == "fallback":
        # Log or handle fallback scenario
        return {"warning": "Used fallback", "result": result["result"]}
    
    return result
